feed-integration-webapp/src/main/java/org/bahmni/module/feedintegration/apicalls/CraterAPIcalls.java

```diff
@@ -0,0 +1,238 @@
+                .setUri(new URI(properties.getProperty("crater.url") + "customers"))
+                .addParameter("contact_name", uuid)
+                .build();
+
+        list_customers.addHeader("Authorization", "Bearer " + auth);
+        list_customers.addHeader("Accept", "*/*");
+        list_customers.addHeader("Content-Type", "*/*");
+        list_customers.addHeader("company", "1");
+
+        CloseableHttpResponse response = httpclient.execute(list_customers);
+        JSONObject myObject = new JSONObject(EntityUtils.toString(response.getEntity()));
+
+        if (myObject.getJSONArray("data").length() == 0) {
+            return "Customer not found";
+
+        } else {
+            return myObject.getJSONArray("data").getJSONObject(0).getString("id");
+        }
+
+    }
+
+    public String del_customer(String auth, OpenMRSPatientFullRepresentation patient) throws JSONException, IOException, URISyntaxException {
+
+        String id = this.get_list_customers("asdasd");
+
+        if (id.equals("Customer not found")) {
+            return id;
+        }
+
+        URL url = new URL("https://demo.craterapp.comapi/v1/customers/delete");
+        HttpURLConnection con = (HttpURLConnection) url.openConnection();
+        con.setRequestMethod("POST");
+        con.setRequestProperty("Authorization", "Bearer " + auth);
+        con.setRequestProperty("Content-Type", "application/json; utf-8");
+        con.setRequestProperty("Accept", "application/json");
+        con.setDoOutput(true);
+        con.setDoInput(true);
+
+        JSONArray ids = new JSONArray();
+        ids.put(id);
+
+        System.out.println(ids);
+        JSONObject parameters = new JSONObject();
+        parameters.put("ids", id);
+        String jsonInputString = parameters.toString();
+
+        System.out.println(jsonInputString);
+
+        try (OutputStream os = con.getOutputStream()) {
+            byte[] input = jsonInputString.getBytes("utf-8");
+            os.write(input, 0, input.length);
+        }
+        con.getInputStream();
+        return "Customer deleted";
+    }
+
+    public void create_update(OpenMRSPatientFullRepresentation patientFR) throws Exception {
+        if (login_verification(auth) == "Authenticated") {
+            String uuid = getuuid(patientFR);
+            if (get_list_customers(uuid).equals("Customer not found")) {
+                create_customer(patientFR);
+            } else {
+                update_customer(patientFR);
+            }
+        }
+        else{
+            System.out.println("Not Authenticated");
+        }
+    }
+
+    public String getuuid(OpenMRSPatientFullRepresentation patientFR) {
+        String uuid = "";
+        for (int i = 0; i <= patientFR.getIdentifiers().size(); i++) {
+            if (patientFR.getIdentifiers().get(i).isPreferred()) {
+                uuid = patientFR.getIdentifiers().get(i).getIdentifier();
+                break;
+            }
+        }
+        return uuid;
```

Comment:
imperative code with mutations... better leverage language features and increase the readability
```
      Optional<OpenMRSPatientIdentifier> uuid1 = patientFR.getIdentifiers().stream()
                .filter(OpenMRSPatientIdentifier::isPreferred)
                .findFirst();

        return uuid1.isPresent() ? uuid1.get().getIdentifier() : EMPTY_STRING;
```